api/src/main/kotlin/edu/wgu/osmt/csv/CsvResource.kt

```diff
@@ -32,6 +36,25 @@ abstract class CsvResource<T>(val debugName: String) {
         return stringWriter.toString()
     }
 
+    fun writeCsvToOutputStream(response: OutputStream, data : Stream<T>): OutputStream? {
+
+        val csvList = toCsv(data.toList())
```

Comment:
Why the data.toList()? Is there any reason to not use a map operation from the stream?

Reply:
we are calling the toCsv fun here, so we would need to instead of calling that impl, create same logic but with a Stream if possible.